config/config.go

```diff
@@ -295,16 +295,18 @@ func NewDefaultAgentConfig() *AgentConfig {
 		},
 	}
 
+	isContainerized := os.Getenv("DOCKER_STS_AGENT") != ""
 	// Set default values for proc/sys paths if unset.
 	// Don't set this is /host is not mounted to use context within container.
 	// Generally only applicable for container-only cases like Fargate.
-	if ddconfig.IsContainerized() && util.PathExists("/host") {
+	if isContainerized && util.PathExists("/host") {
 		if v := os.Getenv("HOST_PROC"); v == "" {
 			os.Setenv("HOST_PROC", "/host/proc")
 		}
 		if v := os.Getenv("HOST_SYS"); v == "" {
 			os.Setenv("HOST_SYS", "/host/sys")
 		}
+		ddconfig.Datadog.SetDefault("container_cgroup_root", "/host/sys/fs/cgroup/")
```

Comment:
In the agent there is a condition on the existence of the folder. Maybe we could extract that to a function in the agent and use the function here. Or just use the same logic.
```suggestion
		if pathExists("/host/sys/fs/cgroup/") {
			config.SetDefault("container_cgroup_root", "/host/sys/fs/cgroup/")
		} else {
			config.SetDefault("container_cgroup_root", "/sys/fs/cgroup/")
		}
```